### PR TITLE
Refactor Uno.Core

### PR DESCRIPTION
**Pull request by carlh98**
GitHub Issue: #

## What is the new behavior?

Refactor Uno.Core by removing the package and adding the necessary classes
This will fix an issue where Uno.Core 4.0.0 has Uno.Core.Extensions dependencies that causes problem with the Umbrella Uno 4 update

## Checklist

Please check if your PR fulfills the following requirements:

- [ ] Documentation has been added/updated
- [ ] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

**Review by JeepLavoie — APPROVED** [review]

File: src/GeneratedSerializers.Generator/UnoCore/ActionAsync.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ******************************************************************
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GeneratedSerializers.UnoCore
+{
+	public delegate Task ActionAsync(CancellationToken ct);
```
Comment (MatFillion) [inline src/GeneratedSerializers.Generator/UnoCore/ActionAsync.cs]: I think everything in what we are bringing here should be internal, we don't want this generatedserializer to expose it.

**Comment by MatFillion** [comment]
I think let's not use any "unocore" words in classes, folders or namespaces, you're removing the dependency so let's have things in a helper folder, and nothing public (should be internal or private)

**Comment by carlh98** [comment]
@MatFillion made the changes requested

File: src/GeneratedSerializers.Generator/Helpers/ActionAsync.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ******************************************************************
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GeneratedSerializers
+{
+	public delegate Task ActionAsync(CancellationToken ct);
```
Comment (MatFillion) [inline src/GeneratedSerializers.Generator/Helpers/ActionAsync.cs]: Internal

File: src/GeneratedSerializers.Generator/Helpers/Funcs.cs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+		/// <typeparam name="TResult">The return value</typeparam>
+		/// <param name="function">The function to memoize</param>
+		/// <returns>The memoized function</returns>
+		public static Func<TParam1, TParam2, TParam3, TParam4, TParam5, TResult> CreateMemoized<TParam1, TParam2, TParam3, TParam4, TParam5, TResult>(Func<TParam1, TParam2, TParam3, TParam4, TParam5, TResult> function)
+		{
+			return function.AsMemoized();
+		}
+	}
+
+	public static class NullableFuncs<T>
```
Comment (MatFillion) [inline src/GeneratedSerializers.Generator/Helpers/Funcs.cs]: internal

File: src/GeneratedSerializers.Generator/Extensions/FuncMemoizeExtensions.cs [hunk 4]
```diff
@@ -29,6 +25,195 @@ namespace GeneratedSerializers.Extensions
 {
 	public static class FuncMemoizeExtensions
```
Comment (MatFillion) [inline src/GeneratedSerializers.Generator/Extensions/FuncMemoizeExtensions.cs]: this whole class should be internal as well no?

**Review by MatFillion — APPROVED** [review]

**Pull request merged**